src/main/java/com/android/volley/toolbox/DiskBasedAsyncCache.java

```diff
@@ -161,28 +181,102 @@ public void completed(Integer resultLen, Void ignore) {
                                 deleteFile(file);
                             }
 
-                            callback.onPutComplete();
+                            callback.onComplete();
                         }
 
                         @Override
                         public void failed(Throwable throwable, Void ignore) {
                             VolleyLog.e(
                                     throwable, "Failed to read file %s", file.getAbsolutePath());
                             deleteFile(file);
-                            callback.onPutComplete();
+                            callback.onComplete();
                             closeChannel(afc, "failed read");
                         }
                     });
         } catch (IOException e) {
             if (closeChannel(channel, "IOException")) {
                 deleteFile(file);
             }
-            callback.onPutComplete();
+            callback.onComplete();
+        }
+    }
+
+    /** Clears the cache. Deletes all cached files from disk. */
+    @Override
+    public void clear(OnCompleteCallback callback) {
+        File[] files = mRootDirectorySupplier.get().listFiles();
+        if (files != null) {
+            for (File file : files) {
+                deleteFile(file);
+            }
+        }
+        mEntries.clear();
+        mTotalSize = 0;
+        VolleyLog.d("Cache cleared.");
+        callback.onComplete();
+    }
+
+    /** Initializes the cache. */
+    @Override
+    public void initialize(OnCompleteCallback callback) {
+        File rootDirectory = mRootDirectorySupplier.get();
+        if (!rootDirectory.exists()) {
+            if (!rootDirectory.mkdirs()) {
+                VolleyLog.e("Unable to create cache dir %s", rootDirectory.getAbsolutePath());
+            }
+            callback.onComplete();
+            return;
+        }
+        File[] files = rootDirectory.listFiles();
+        if (files == null) {
+            callback.onComplete();
+            return;
+        }
+        for (File file : files) {
+            Path path = file.toPath();
+            AsynchronousFileChannel afc = null;
+            try {
+                afc = AsynchronousFileChannel.open(path, StandardOpenOption.READ);
+                int entrySize = (int) file.length();
+                CacheHeader entry = CacheHeader.readHeader(afc, entrySize);
```

Comment:
Similarly, here we'll want to kick off asynchronous reads of each file (ideally in parallel), and as the results come in, we should populate our map, rather than doing this effectively sequentially and in a blocking manner. CompletableFuture.allOf would likely help here.